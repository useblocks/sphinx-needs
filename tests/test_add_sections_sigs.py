import json
import re
from pathlib import Path

import pytest
from sphinx.testing.util import SphinxTestApp


@pytest.mark.parametrize(
    "test_app",
    [
        {
            "buildername": "html",
            "srcdir": "doc_test/add_sections_sigs",
            "no_plantuml": True,
        }
    ],
    indirect=True,
)
def test_section_is_usable_in_filters(test_app: SphinxTestApp, snapshot):
    app = test_app
    app.build()
    assert not app._warning.getvalue()

    needs_data = json.loads(Path(app.outdir, "needs.json").read_text())
    assert needs_data["versions"][""]["needs"] == snapshot

    html = Path(app.outdir, "index.html").read_text()

    tables = re.findall("(<table .*?</table>)", html, re.DOTALL)
    assert len(tables) == 5

    # All requirements should be in first table
    assert "R_12345" in tables[3]
    assert "First Section" in tables[3]
    assert "R_12346" in tables[3]
    assert "Second Section" in tables[3]

    # Only requirements from the first section should be in table 2
    assert "R_12345" in tables[4]
    assert "First Section" in tables[4]
    assert "R_12346" not in tables[4]
    assert "Second Section" not in tables[4]
