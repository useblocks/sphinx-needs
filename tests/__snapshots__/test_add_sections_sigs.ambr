# serializer version: 1
# name: test_section_is_usable_in_filters[test_app0]
  dict({
    'R_12345': dict({
      'content': 'The Tool **shall** have a command line interface.',
      'docname': 'index',
      'external_css': 'external_link',
      'full_title': 'Command line interface',
      'id': 'R_12345',
      'impacts': 'component_a',
      'introduced': '1.0.0',
      'layout': '',
      'lineno': 9,
      'section_name': '1.1 First Section',
      'sections': list([
        '1.1 First Section',
        '1 TEST DOCUMENT',
      ]),
      'tags': list([
        'test',
        'test2',
      ]),
      'title': 'Command line interface',
      'type': 'req',
      'type_name': 'Requirement',
      'updated': '1.5.1',
    }),
    'R_12346': dict({
      'content': 'The Tool **shall** have a command line interface.',
      'docname': 'index',
      'external_css': 'external_link',
      'full_title': 'Another Requirement',
      'id': 'R_12346',
      'impacts': 'component_b',
      'introduced': '1.1.1',
      'layout': '',
      'lineno': 20,
      'section_name': '1.2 Second Section',
      'sections': list([
        '1.2 Second Section',
        '1 TEST DOCUMENT',
      ]),
      'tags': list([
        'test',
        'test2',
      ]),
      'title': 'Another Requirement',
      'type': 'req',
      'type_name': 'Requirement',
      'updated': '1.4.0',
    }),
    'T_001': dict({
      'docname': 'index',
      'external_css': 'external_link',
      'full_title': 'test method',
      'id': 'T_001',
      'layout': '',
      'lineno': 40,
      'section_name': '1.3 need in API',
      'sections': list([
        '1.3 need in API',
        '1 TEST DOCUMENT',
      ]),
      'signature': 'test_method',
      'title': 'test method',
      'type': 'test',
      'type_name': 'Test Case',
    }),
  })
# ---
